src/node.cc

```diff
@@ -2661,9 +2712,24 @@ void SetupProcessObject(Environment* env,
 
   process->SetAccessor(env->title_string(),
                        ProcessTitleGetter,
-                       ProcessTitleSetter,
+                       env->is_main_instance() ? ProcessTitleSetter : nullptr,
                        env->as_external());
 
+  // process.threadId
+  READONLY_PROPERTY(process,
+                    "threadId",
+                    Number::New(env->isolate(), env->thread_id()));
+
+  // process.isMainInstance
+  READONLY_PROPERTY(process,
+                    "isMainInstance",
+                    Boolean::New(env->isolate(), env->is_main_instance()));
+
+  // process.isWorkerInstance
+  READONLY_PROPERTY(process,
+                    "isWorkerInstance",
+                    Boolean::New(env->isolate(), env->is_worker_instance()));
```

Comment:
I'm fine with having both. Matches how `cluster` works. This is a nit, but possibly consider how to shorten the names, or maybe just swap `Instance` for `Thread`.